.devcontainer/Dockerfile

```
@@ -0,0 +1,18 @@
+FROM mcr.microsoft.com/devcontainers/python:3.9-bullseye
+
+USER vscode
+
+# install rust
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+
+# install the linker
+RUN sudo apt-get update && export DEBIAN_FRONTEND=noninteractive && sudo apt-get -y install --no-install-recommends build-essential gcc 
```

```suggestion
USER root
RUN apt-get update && export DEBIAN_FRONTEND=noninteractive && apt-get -y install --no-install-recommends build-essential gcc && rm -rf /var/lib/apt/lists/*

USER vscode
```

Ensure remove `/var/lib/apt/lists/*` after `apt update` to reduce image size.

In addition, if `sudo` is used, the command will fail behind the proxy because the environment variable is not propagated.